### PR TITLE
fix: one and only one front

### PR DESCRIPTION
**Pull request by ivarconr**
Still need to fix:

- [x] /projects loaded all over the place. Fixed by only loading it when `flags=P` is set.  
- ~[ ] /contexts should also just return "default context fields", but not allow PUT/POST/DELETE. Can probably fix this by having a route only allowing GET and the others part of enterprise.~ (not relevant for this PR)
- [x] favicon needs to be update one (enterprise)
- [x] Logo should be the enterprise logo

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
```
Comment (ivarconr) [inline .github/workflows/nodejs.yml:1]: TODO: remove this,

File: package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "unleash-frontend",
+  "name": "@ivarconr/unleash-enterprise-frontend",
```
Comment (ivarconr) [inline package.json:2]: revert this!

File: src/component/context/form-context-component.jsx [hunk 3]
```diff
@@ -17,7 +25,7 @@ class AddContextComponent extends Component {
     }
 
     static getDerivedStateFromProps(props, state) {
-        if (!state.contextField.name && props.contextField.name) {
+        if (state.contextField.inital && !props.contextField.inital) {
```
Comment (chriswk) [inline src/component/context/form-context-component.jsx:28]:
```suggestion
        if (state.contextField.initial && !props.contextField.initial) {
```

File: src/component/context/edit-context-container.js [hunk 4]
```diff
@@ -6,6 +6,9 @@ const mapStateToProps = (state, props) => {
     const contextFieldBase = { name: '', description: '', legalValues: [] };
     const field = state.context.toJS().find(n => n.name === props.contextFieldName);
     const contextField = Object.assign(contextFieldBase, field);
+    if (!field) {
+        contextField.inital = true;
```
Comment (chriswk) [inline src/component/context/edit-context-container.js:10]:
```suggestion
        contextField.initial = true;
```

File: src/component/feature/project-select-component.jsx [hunk 5]
```diff
@@ -4,16 +4,16 @@ import MySelect from '../common/select';
 
 class ProjectSelectComponent extends Component {
     componentDidMount() {
-        const { fetchProjects, projects } = this.props;
-        if (projects[0].inital && fetchProjects) {
-            this.props.fetchProjects();
+        const { fetchProjects, projects, enabled } = this.props;
+        if (projects[0].inital && enabled) {
```
Comment (chriswk) [inline src/component/feature/project-select-component.jsx:8]:
```suggestion
        if (projects[0].initial && enabled) {
```

File: src/page/admin/auth/google-auth-container.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { connect } from 'react-redux';
+import GooleAuth from './google-auth';
```
Comment (FredrikOseberg) [inline src/page/admin/auth/google-auth-container.js:2]:
```suggestion
import GoogleAuth from './google-auth';
```

File: src/page/admin/auth/google-auth-container.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+import { getGoogleConfig, updateGoogleConfig } from './../../../store/e-admin-auth/actions';
+import { hasPermission } from '../../../permissions';
+
+const mapStateToProps = state => ({
+    config: state.authAdmin.get('google'),
+    hasPermission: permission => hasPermission(state.user.get('profile'), permission),
+});
+
+const Container = connect(mapStateToProps, { getGoogleConfig, updateGoogleConfig })(GooleAuth);
```
Comment (FredrikOseberg) [inline src/page/admin/auth/google-auth-container.js:11]:
```suggestion
const Container = connect(mapStateToProps, { getGoogleConfig, updateGoogleConfig })(GoogleAuth);
```

File: src/page/admin/auth/google-auth.jsx [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+        getGoogleConfig();
+    }, []);
+
+    useEffect(() => {
+        if (config.clientId) {
+            setData(config);
+        }
+    }, [config]);
+
+    if (!hasPermission('ADMIN')) {
```
Comment (FredrikOseberg) [inline src/page/admin/auth/google-auth.jsx:25]: Not relevant for this PR, but should we extract permission types into constants?

File: src/component/menu/routes.js [hunk 9]
```diff
@@ -58,12 +63,18 @@ export const routes = [
...
     { path: '/context/create', parent: '/context', title: 'Create', component: CreateContextField },
     { path: '/context/edit/:name', parent: '/context', title: ':name', component: EditContextField },
-    { path: '/context', title: 'Context Fields', icon: 'apps', component: ContextFields, hidden: true },
+    { path: '/context', title: 'Context Fields', icon: 'album', component: ContextFields, flag: 'C' },
 
     // Project
     { path: '/projects/create', parent: '/projects', title: 'Create', component: CreateProject },
     { path: '/projects/edit/:id', parent: '/projects', title: ':id', component: EditProject },
-    { path: '/projects', title: 'Projects', icon: 'folder_open', component: ListProjects, hidden: true },
+    { path: '/projects', title: 'Projects', icon: 'folder_open', component: ListProjects, flag: 'P' },
```
Comment (FredrikOseberg) [inline src/component/menu/routes.js:71]: Consider extracting flag types into constants.

**Review by FredrikOseberg — APPROVED** [review]
LGTM. I think the flag approach works, and is a good first step. We could consider refactoring later to a central class or context that holds the flags state and delegates methods instead.

File: src/page/admin/auth/google-auth.jsx (hunk 8 above)
Comment (ivarconr) [reply src/page/admin/auth/google-auth.jsx:25]: yes, we need to soon fix the permission systems anyways to also support project specific permissions.

**Comment by ivarconr** [comment]
> LGTM. I think the flag approach works, and is a good first step. We could consider refactoring later to a central class or context that holds the flags state and delegates methods instead.

Yeah, I think a context would be a good fit for flags.

**Pull request merged**